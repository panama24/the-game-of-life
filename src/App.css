.App {
  text-align: center;
  padding: 64px 128px;
}

.grid-container {
  border: 1px solid red;
  display: grid;
  grid-template-columns: repeat(30, 1fr);
  direction: rtl;
}

.item {
  border: 1px solid red;
}
