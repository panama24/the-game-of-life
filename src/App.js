import React from 'react';
import logo from './logo.svg';
import './App.css';

export const trueOrFalse = () => !!Math.floor(Math.random() * 2);
export const generateRandArray = () => Array(10).fill(null).map(c => trueOrFalse());

export const generateGrid = () => Array(10)
  .fill(generateRandArray())
  .map(arr => generateRandArray());

export const shouldLive = (cellState, neighbors) => {
  if (cellState) {
    const count = neighbors.reduce((acc, curr) => {
      if (curr) {
        acc += 1
      }
      return acc;
    }, 0);

    return count === 2 || count === 3;
  }

  if (!cellState) {
    const count = neighbors.reduce((acc, curr) => {
      if (curr) {
        acc += 1
      }
      return acc;
    }, 0);

    return count === 3;
  }
};

export const perfectCond = (grid, rowIdx, colIdx) => {
  const row = grid[rowIdx];
  const prevRow = grid[rowIdx-1];
  const nextRow = grid[rowIdx+1];

  const cell = row[colIdx];
  const prevCell = row[colIdx-1];
  const nextCell = row[colIdx+1];

  if (row === 0) {
    if (cell === 0) {
      // is top left corner
      const check = [nextCell, nextRow[colIdx], nextRow[colIdx+1]]
    }
    if (cell === row.length-1) {
      // is top right corner
      const check = [prevCell, nextRow[colIdx], nextRow[colIdx-1]]
    }
  }

  if (row === grid.length -1) {
    if (cell === 0) {
      // is bottom left corner
      const check = [nextCell, prevRow[colIdx], prevRow[colIdx+1]]
    }
    if (cell === row.length-1) {
      // is bottom right corner
      const check = [prevCell, prevRow[colIdx], prevRow[colIdx-1]]
    }
  }

  // is a middle cell
  const check = [
    prevRow[colIdx-1],
    prevRow[colIdx],
    prevRow[colIdx+1],
    prevCell,
    nextCell,
    nextRow[colIdx-1],
    nextRow[colIdx],
    nextRow[colIdx+1],
  ];
};

export class Cell {
  isAlive() {
    return true;
  }
}
// export const Grid = () => {
  // const arrays = generateGrid();
  // return (
    // <>
      // {arrays.map((a, i) => {
        // return (
          // <Cell key={i} content={a} />
        // )
      // })}
    // </>
  // );
// };

function App() {
  return (
    <div className="App">
    </div>
  );
}

export default App;
