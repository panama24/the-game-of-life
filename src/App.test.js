import React from 'react';
import ReactDOM from 'react-dom';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom';
import App, { generateGrid, Cell, trueOrFalse } from './App';

afterEach(cleanup);

/*
 * Alive cell - if fewer than 2 alive neighbors, dies due to underpopulation)
 * Alive cell - 2 or 3 neighbors, lives due to perfect situation
 * Alive cell - more than 3 alive neighbors, dies due to overpopulation
 * Dead cell - exactly 3 alive neighbors, becomes alive due to reproduction
 */

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

describe('App', () => {
  it('should match snapshot', () => {
    const { asFragment } = render(<App />);
    expect(asFragment()).toMatchSnapshot();
  });
});

describe('generateGrid', () => {
  it('should be a 10 x 10 grid', () => {
    const grid = generateGrid();
    const rows = grid.length;
    /* [
     * [0,0,0,0,0,0,0,0,0,0]
     * [0,0,0,0,0,0,0,0,0,0]
     * [0,0,0,0,0,0,0,0,0,0]
     * [0,0,0,0,0,0,0,0,0,0]
     * [0,0,0,0,0,0,0,0,0,0]
     * [0,0,0,0,0,0,0,0,0,0]
     * ] */;
    const columns = grid.every(r => r.length === 10);
    expect(rows).toEqual(10);
    expect(columns).toEqual(true);
  });
});

describe('Cell', () => {
  it('should die if has fewer than 2 living neighbors', () => {
    /*
    prev row = i-1, i, i+1
    same row = i-1, i, i+1
    next row = i-1, i, i+1
    */
  });

  it('should remain alive if has 2-3 living neighbors', () => {
  });

  it('should die if has more than 3 living neighbors', () => {
  });

  it('should die if has more than 3 living neighbors', () => {
  });
});

describe('trueOrFalse', () => {
  it('chooses a random boolean', () => {
    const res = [...Array(10)]
      .fill(null)
      .map(el => trueOrFalse());
    const expectation = res.every(el => el);
    expect(expectation).toBe(false);
  });
});
