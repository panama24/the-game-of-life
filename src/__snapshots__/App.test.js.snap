// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="App"
  />
</DocumentFragment>
`;
